src/realm/object-store/c_api/query.cpp

```diff
@@ -256,11 +255,15 @@ RLM_API realm_query_t* realm_query_parse_for_results(const realm_results_t* resu
                                                      size_t num_args, const realm_query_arg_t* args)
 {
     return wrap_err([&]() {
+        auto existing_query = results->get_query();
         auto realm = results->get_realm();
         auto table = results->get_table();
-        Query query = parse_and_apply_query(realm, table, query_string, num_args, args);
-        auto ordering = query.get_ordering();
-        return new realm_query_t{std::move(query), std::move(ordering), realm};
+        auto query = parse_and_apply_query(realm, table, query_string, num_args, args);
```

Comment:
I think you are losing the ordering, should you not use `combined.get_orderning()` ? Also, we need to add testing around this…

Reply:
I don't think we should be merging the ordering when using the query parser API. e.g. what is your expectation for the order of `realm.all<Person>("age > 18 SORT(firstName desc)").filter("TRUEPREDICATE SORT(lastName desc)")`?

I think the second ordering should replace the first one. If we combine orderings, then there's no way for the developer to remove the ordering, whereas if they want a complex sort order, they can always do it by providing it in the last predicate to run.

Reply:
ok it makes sense, we need some testing around this API

Reply:
It might be worth exposing `SortDescriptor::MergeMode` here to allow users to choose what happens depending on how they build up their queries.

Reply:
I think we probably want to do it with a separate API - one that only controls sorting. Something like:

```dart
realm.all<Person>()
    .filter("lastName STARTSWITH 'a'")
    .sort('firstName', order: SortOrder.ascending)
    .sort('lastName', order: SortOrder.ascending, mergeMode: SortMergeMode.append);
```

Reply:
I agree, there shold be another API.